web-jehlomat/src/screens/Nalezy/Components/Table.tsx

```diff
@@ -32,42 +30,47 @@ const Header = styled('tr')({
     width: '100%',
 });
 
-const Table: FunctionComponent<TableProps> = ({ loader, direction, onSort, selected, onSelect, onUpdate }) => {
+const Table: FunctionComponent<TableProps> = ({ loader }) => {
+    const setSort = useSetRecoilState(sortingState);
+
     const loading = loader.resp === undefined && loader.err === undefined;
     const error = loader.resp === undefined && loader.err !== undefined;
     const loaded = loader.resp !== undefined;
     const data = loader.resp?.syringeList || [];
+    const pageInfo = loader.resp?.pageInfo || { size:0, index: 0 };
+
+    const handleSort = (column: SortableColumn) => () => setSort(state => sort(state, column));
 
     return (
         <Container>
             <Wrapper>
                 <thead>
                     <Header>
                         <Heading />
-                        <SortableHeading direction={direction('TOWN')} onClick={onSort('TOWN')}>
-                            Město
+                        <SortableHeading direction={useRecoilValue(columnSortingDirection('TOWN'))} onClick={handleSort('TOWN')}>
```

Comment:
What about creating array with directions and loop over it here?

Reply:
Parret principle, it would be too complex for such a simple thing.